scripts/pythondistdeps.py

```diff
@@ -334,25 +371,22 @@ def normalize_name(name):
                     spec = ('==', dist.py_version)
                     if spec not in py_deps[name]:
                         py_deps[name].append(spec)
-                deps = dist.requires()
+                deps = dist.requirements
                 if args.recommends:
-                    depsextras = dist.requires(extras=dist.extras)
-                    if not args.requires:
-                        for dep in reversed(depsextras):
-                            if dep in deps:
-                                depsextras.remove(dep)
-                    deps = depsextras
+                    deps = dist.extra_requirements
                 # console_scripts/gui_scripts entry points need pkg_resources from setuptools
-                if ((dist.get_entry_map('console_scripts') or
-                    dist.get_entry_map('gui_scripts')) and
+                if (dist.entry_points and
                     (lower.endswith('.egg') or
                      lower.endswith('.egg-info'))):
-                    # stick them first so any more specific requirement overrides it
-                    deps.insert(0, Requirement.parse('setuptools'))
+                    groups = set([ep.group for ep in dist.entry_points])
+                    if set(["console_scripts", "gui_scripts"]) & groups:
+                        # stick them first so any more specific requirement
+                        # overrides it
+                        deps.insert(0, Req('setuptools'))
                 # add requires/recommends based on egg/dist metadata
                 for dep in deps:
                     if normalized_names_require_pep503:
-                        dep_normalized_name = normalize_name(dep.project_name)
+                        dep_normalized_name = normalize_name(dep.name)
                     else:
                         dep_normalized_name = dep.key
```

Comment:
@hroncok But the code continues to preserve both the legacy and pep503 conventions. So I'm not sure I understand what you mean.

Reply:
In that case, I don't get the question :)

Reply:
> Be careful with manual replaces for normalized names. There are two regexes that this code supported before. The PEP 503 regex and the regex imported from setuptools (described in here).

I wasn't exactly sure what you meant by this comment, so I asked if the new code looks good to you to be sure.